### PR TITLE
version 0.128, various fixes and enhancements

### PR DESCRIPTION
**Pull request by twessling-icas**
* added multi-campaign documentation, only available for Kleinanzeigen now though.
* extra explanation on the 'vendorId' concept
* extra explanation on `_include` and `_exclude`
* fixes on some examples
* added vendorId, title and isDefault to campaign V5 struct
* more explanation on usage of pageToken
* added some error codes

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -1721,6 +1729,8 @@ paths:
             - 'console_ro'
             - 'api_ro'
         # FEEDS
```
Comment (remenska) [inline openapi/openapi.yaml]: minor: this instead of L1733

File: src/source/ads.rst [hunk 2]
```diff
@@ -137,7 +136,8 @@ externalId
 
 Any non-empty string with a maximum length of 64 characters. When fetching
 an existing ad which does not have an **externalId**, the field is empty or
-omitted.
+omitted. There is no unique constraint when it comes to externalId, so multiple
+ads are allowed to have the same externalId.
```
Comment (remenska) [inline src/source/ads.rst:140]: maybe update the yaml doc as well, on this field.

File: src/source/ads.rst [hunk 3]
```diff
@@ -254,9 +254,8 @@ campaignId
 """"""""""
 
 identifier of the campaign this ad belongs to. See :ref:`campaigns` for more info.
-For the time being this is not mandatory, and we will add the ad to the one-and-only campaign the seller has.
-Once we will start allowing multiple campaigns for a seller, we will change this to become mandatory.
-For the time being, we will also create the campaign underwater if the seller has no campaign created yet, to ensure
+This is not mandatory, and we will add the ad to the `default campaign` if not provided.
+For the time being, we will also create the (default) campaign underwater if the seller has no campaign created yet, to ensure
```
Comment (remenska) [inline src/source/ads.rst:258]: same, worth adjusting the yaml

File: src/source/ads.rst [hunk 4]
```diff
@@ -561,6 +560,9 @@ Any non-empty string with a maximum length of 64 characters. Should be unique
 per customer. Can either be set when creating an ad or when updating an
 existing ad. However, once set, it can no longer be modified. When fetching an
 existing ad which does not have a **vendorId**, the field is omitted.
+When an ad gets set to status ``DELETED``, this does **not** remove the vendorId.
```
Comment (remenska) [inline src/source/ads.rst:563]: ditto

File: src/source/api-reference.rst [hunk 5]
```diff
@@ -43,10 +43,11 @@ API Reference
    reference/post-metrics-data
    reference/campaign-status
    reference/post-campaign
-   reference/get-campaign
+   reference/get-campaigns
```
Comment (remenska) [inline src/source/api-reference.rst:46]: not worth updating this, the whole file should be removed and not referenced anywhere. (I did remove it from index iirc)

File: src/source/examples/get-ad-v5-example.rst [hunk 6]
```diff
@@ -40,6 +40,7 @@
           "dateCreated": "2012-08-31T16:12:53Z",
           "dateLastUpdated": "2012-09-05T04:03:42Z",
           "vendorId": "OURSHOP-1423453-34",
+          "campaignId": 124152,
```
Comment (remenska) [inline src/source/examples/get-ad-v5-example.rst:43]: same, I don't think this shows anywhere in our doc anymore (unless it's directly referenced in our freeform part, rather than the API reference)

File: src/source/examples/get-campaign-vendor-id-v5-example.rst [hunk 7]
```diff
@@ -0,0 +1,27 @@
+.. code-block:: javascript
```
Comment (remenska) [inline src/source/examples/get-campaign-vendor-id-v5-example.rst:1]: this already should exist in the yaml, worth double checking if it's correct though

File: src/source/reference/get-ad.rst [hunk 8]
```diff
@@ -31,7 +31,8 @@ GET /ad v5
    - ``application/sellside.ad.list-v5+json, application/json``
 
 :ref:`get_ad_v5` is quite similar to :ref:`get_ad_v3` and :ref:`get_ad_v4`, except for some changes in the :ref:`ads` and an addition of a ``nextPageToken`` field in the response struct.
-This ``nextPageToken`` replaces the usage of the 'offset' parameter and allows the caller to paginate through a large list of ads more efficiently.
+The value of this ``nextPageToken`` should be used as *additional* parameter `pageToken` to your regular call and replaces the usage of the 'offset' parameter.
+This approach allows the caller to paginate through a large list of ads much more efficiently.
```
Comment (remenska) [inline src/source/reference/get-ad.rst:35]: this also needs to go in yaml

**Review by remenska — COMMENTED** [review]
I think I'll create a follow up removal of anything under `/reference/*`, I see it still creates confusion

File: src/source/overview.rst [hunk 9]
```diff
@@ -115,7 +115,7 @@ header to::
 Image Downloads
 ---------------
 
-Images are downloaded from the following ip addresses. Ensure that all image
+Images & feeds are downloaded from the following ip addresses. Ensure that all image
```
Comment (jayuuza) [inline src/source/overview.rst:118]: Tiny nitpick. Found this sentence a little confusing.
Would rephrase it:
```
Images & Feed imports are downloaded via the following ip addresses.
```

**Review by jayuuza — APPROVED** [review]
